src/frontend/src/lib/types/token.ts

```diff
@@ -20,7 +22,16 @@ export interface TokenMetadata {
 	icon?: string;
 }
 
-export type RequiredToken = Required<Token>;
+export interface TokenAppearance {
+	displayName?: TokenDisplayName;
+}
+
+export type TokenDisplayName = {
+	prefix: string | undefined;
+	name: string;
+};
+
+export type RequiredToken = RequiredExcept<Token, keyof TokenAppearance>;
```

Comment:
Could we implement standard TypeScript in this pull request by using constructs like `Omit<Token, "displayName">`, and then, in a subsequent PR, doing a refactoring and introducing TypeScript utilities?

i.e. implementing the feature first and then refactoring, rather than doing everything within the same PR, given that we have already iterated on this solution multiple times.

Reply:
ah yes, sure!

Reply:
- PR #1985